### PR TITLE
(ft-client-logout-182201571):Logout Functionality

### PR DESCRIPTION
**Pull request by ishimweeli**
**What does this PR do?**
Users should be able to logout  to Barefoot Nomad

**### How should this be manually tested?**
After cloning this repository, Switch to this branch
The run: yarn install  
To run the app: yarn start  
then send logout request
Any background context you want to provide?

What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/182201571

### CODE REVIEW

**Review by kalisaNkevin — DISMISSED** [review]
Import api from the utils , and also there are no tests and lastly i would recommend you to use redux/toolkit.

File: src/components/auth/auth.logout.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  function handleLogout(){
+
+axios.get(url,{headers:{jwt:"token"}})
+.then((res) => {
+  if (res.status === 200) {
+
+   sweerAlt.fire(`Hey`, 'Welcome to Barefoot Nomad', 'success');
+   navigate('/')
+  }else{
+   sweerAlt.fire('Oops...', `${err.response.data.message}`, 'error');
```
Comment (ericus123) [inline src/components/auth/auth.logout.js]: You should not use sweet alert. 
Use material UI alerts

File: src/components/auth/auth.logout.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+const Logout=()=>{
+    
+  
+  const navigate= useNavigate()
+  function handleLogout(){
+
+axios.get(url,{headers:{jwt:"token"}})
+.then((res) => {
+  if (res.status === 200) {
```
Comment (ericus123) [inline src/components/auth/auth.logout.js]: status check here is unnecessary.

File: src/components/auth/auth.logout.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import React, { useCallback } from 'react'
+import { Grid,Paper, Avatar, TextField, Button, Typography,Link } from '@material-ui/core'
+import sweerAlt from'sweetalert2'
+
+import axios from 'axios'
+import { useNavigate } from 'react-router-dom'
+const url = "https://elite-bn-be.herokuapp.com/api/v1/user/logout"
```
Comment (ericus123) [inline src/components/auth/auth.logout.js]: Why do you have this hardcoded ?????

**Review by ericus123 — CHANGES_REQUESTED** [review]

File: src/components/auth/auth.logout.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    return(
+        <Grid>
+           
+   <Button type='submit' color='primary' variant="contained"  onClick={handleLogout} style={btnstyle}>Logout</Button>
+
+   {openError&&<Alert  onClose={handleClose} severity="error" style={alertStyle}>Oops failed please try again!</Alert>}
+
+   {openSuccess&&<Alert severity="success" onClose={handleClose} >Hey  — you have sucessful loggedout !</Alert>}
+        </Grid>
+    )
```
Comment (ericus123) [inline src/components/auth/auth.logout.js]: No need for alert on logout

**Comment by ericus123** [comment]
This is what I'm getting.

**Review by ericus123 — DISMISSED** [review]

**Comment by ericus123** [comment]
@ishimweeli rebase

**Comment by didasmbalanya** [comment]
Please fix failing tests so that it can be tested

**Comment by didasmbalanya** [comment]
Logout button does not seem to work on the review app. No action on click

**Comment by ishimweeli** [comment]
> Logout button does not seem to work on the review app. No action on click

I have fixed all things mentioned above

File: src/components/auth/auth.logout.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    .catch(error=>{console.log(error)})
+}
+  
+    // const handleClose = (event, reason) => {
+    //       if (reason === 'clickaway') {
+    //         return;
+    //       }
+      
+    //       setOpenError(false);
+    //     };
```
Comment (ericus123) [inline src/components/auth/auth.logout.js]: remove this

File: src/components/auth/auth.logout.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+ const [openSuccess,setOpenSuccess]=useState(false)
+ const [openError,setOpenError]=useState(false)
+ 
+function handleLogout(){
+  const jwt=localStorage.getItem("jwt")
+  api.get('api/v1/user/logout',{headers:{jwt}})
+    .then((res) => {
+      localStorage.removeItem('jwt');
+
+      console.log(res)
```
Comment (ericus123) [inline src/components/auth/auth.logout.js]: remove this

**Review by ericus123 — CHANGES_REQUESTED** [review]

**Comment by ericus123** [comment]
Logout should be done only by clicking one button.
Why are you linking it with two buttons.
The button in the nav dropdown is enough.

@ishimweeli fix that ASAP

**Review by ericus123 — APPROVED** [review]

**Pull request merged**